test/tests/advancedSearchTest.js

```diff
@@ -22,7 +22,9 @@ describe("Advanced Search", function () {
 		var promise = waitForWindow('chrome://zotero/content/advancedSearch.xhtml');
 		zp.openAdvancedSearchWindow();
 		var searchWin = yield promise;
-		
+		// After upgrading to react 18, at this point the onload listener has
+		// not finished running yet, so an extra tick needs to be skipped
+		yield Zotero.Promise.delay();
```

Comment:
I'm wondering if it wouldn't be better to wait for something solid, not just the next call stack, just to make sure test won't end up being flaky (it doesn't seem to be so maybe irrelevant). E.g. we could make use of the second argument to `waitForWindow` instead?

````js
var promise = waitForWindow('chrome://zotero/content/advancedSearch.xhtml', async (win) => {
			while (!win.ZoteroAdvancedSearch.itemsView) {
				Zotero.debug("Waiting for ZoteroAdvancedSearch");
				await Zotero.Promise.delay(5);
			}
		});
````